packages/dds/tree/src/simple-tree/treeNodeApi.ts

```diff
@@ -201,26 +201,24 @@ export const treeNodeApi: TreeNodeApi = {
 	},
 	shortId(node: TreeNode): number | string | undefined {
 		const flexNode = getFlexNode(node);
-		let shortId: number | string | undefined;
-		for (const field of flexNode.boxedIterator()) {
-			if (field.schema.kind === FieldKinds.identifier) {
-				if (shortId !== undefined) {
-					throw new UsageError(
-						"shortId() may not be called on a node with more than one identifier. Consider converting extraneous identifier fields to string fields.",
-					);
-				}
-				const identifier = field.boxedAt(0);
+		const identifierFieldKeys = (flexNode.schema as FlexObjectNodeSchema).identifierFieldKeys;
```

Comment:
🚨 CAST ALERT 🚨 

We don't actually know that `flexNode.schema` is a FlexObjectNodeSchema, do we? You can pass any kind of `TreeNode` into `shortId`. Either `shortId` needs to have its parameter narrowed to only object nodes (which would be a breaking change unfortunately), or you need to use `instanceof` to check the schema is of the proper type before trying to read `identifierFieldKeys` off of it.

Since this isn't currently breaking any tests, and it's incorrect, I think it'd be worth adding a test that calls `shortId` with a non-object node too.

Reply:
Whoops, that makes sense :)

Updated to to check using instance of, and added a test case with a non-object node